test/services/host_inventory_api_test.rb

```diff
@@ -51,4 +51,20 @@ class HostInventoryApiTest < ActiveSupport::TestCase
     @api.expects(:create_host_in_inventory).returns(@host)
     assert_equal @host, @api.inventory_host
   end
+
+  test 'find_results matches on ID' do
+    assert_not_nil @api.send(
```

Comment:
🤔 (there is no friendly suspicious looking emoji)

Asserting that it returns a result with the matching `id` instead of `not_nil` would be much more appropiate.

Reply:
This is checking that, given this response with the matching ID as set up on line 12, the find_results method properly matches based on ID. Checking again that the ID matches doesn't actually test any code, as that is exactly what find_results is doing.

Reply:
It's true whatever the `find_results` method returns is irrelevant, as in `host_already_in_inventory` we're not even checking the response, we just return the result if there was some matching, or nil otherwise. 

It'd make sense to me too to at least add the check here that `find_results` is matching the right host, and perhaps also making sure `host_already_in_inventory`  does too, it is not supposed to break any tests.

Reply:
Ok, I've updated these tests to validate the ID or name matches